posydon/binary_evol/binarystar.py

```diff
@@ -389,9 +396,50 @@ def to_df(self, **kwargs):
             # If a binary fails, usually history cols have diff lengths.
             # This should append NAN to create even columns.
             all_equal_length_cols = len(set(col_lengths)) == 1
+
             if not all_equal_length_cols:
-                for col in data_to_save:
-                    col.extend([np.nan] * abs(max_col_length - len(col)))
+                for i, col in enumerate(data_to_save):
+
+                    dkey = all_keys[i]
+                    try:
+                        dtype = properties_dtypes[dkey]
+                    except KeyError:
+                        dtype = ''
+                        pass
```

Comment:
Just a bit of style (one-liner and without raising errors):
```suggestion
                    dtype = properties_dtypes.get(dkey, '')
```